week-1-ts-ecosystem/w01e02-enter-the-world-of-typescript/src/framework/dom-creators.ts

```diff
@@ -1,21 +1,17 @@
-function createElement<Element extends HTMLElement>(
+function createElement<TAG extends HTMLParagraphElement | HTMLDivElement | HTMLElement>(
   tagName = 'div',
   className = ''
-): Element {
-  const element = document.createElement(tagName);
+): TAG {
+  const element = <TAG>document.createElement(tagName);
```

Comment:
TO nie do końca poprawione, ponieważ to inny zapis `as` :) 

Dalej to asercja typów i tego nie kupuje.